### PR TITLE
work around safari hls h265 problem

### PR DESCRIPTION
**Pull request by yaniv-li**
This is a workaround the problem described in the ticket 
https://cloudinary.atlassian.net/browse/APPS-9349

### CODE REVIEW

File: src/plugins/cloudinary/index.js [hunk 1]
```diff
@@ -234,21 +235,35 @@ class CloudinaryContext extends mixin(Playlistable) {
...
+            if (canPlay) {
+              return src;
+            }
           }
         }
-        return true;
+        return src;
       });
+      // filter out undefined
+      _sources.filter(s => s);
```
Comment (e1adn) [inline src/plugins/cloudinary/index.js:257]: please remove it after replace `src.generateSources().map` with reduce

Comment (e1adn) [reply src/plugins/cloudinary/index.js:257]: and 'filter' creates a new array. looks like it does not set the new array to any variable.  I do not completely understand what is it for?

Comment (yaniv-li) [reply src/plugins/cloudinary/index.js]: @e1adn  done

File: src/plugins/cloudinary/index.js [hunk 2]
```diff
@@ -234,21 +235,35 @@ class CloudinaryContext extends mixin(Playlistable) {
         this.player.poster(src.poster().url());
       }
 
-      _sources = src.generateSources().filter((src) => {
+      _sources = src.generateSources().map((src) => {
```
[the remark this answers is not in the document]
Comment (e1adn) [reply src/plugins/cloudinary/index.js]: Screenshot at Jun 06 15-52-18

[the remark this answers is not in the document]
Comment (e1adn) [reply src/plugins/cloudinary/index.js]: @yaniv-li does work on master and edge

**Review by e1adn — APPROVED** [review]

**Pull request merged**